src/app/template/shared/base-template.model.ts

```diff
@@ -1,74 +1,56 @@
 import { Taggable } from '../../shared/interfaces/taggable.interface';
 import { TemplateTagKeys } from '../../shared/services/tags/template-tag-keys';
 import { Utils } from '../../shared/services/utils/utils.service';
+import { Iso } from './iso.model';
+
+
+export interface BaseTemplateModel extends BaseModelInterface, Taggable {
+  id: string;
+  account: string;
+  created: Date;
+  crossZones: boolean;
+  displaytext: string;
+  domain: string;
+  domainid: string;
+  isdynamicallyscalable: boolean;
+  isextractable: boolean;
+  isfeatured: boolean;
+  ispublic: boolean;
+  isready: boolean;
+  name: string;
+  ostypeid: string;
+  ostypename: string;
+  osType: OsType;
+  size: number;
+  status: string;
+  tags: Array<Tag>;
+  zoneid: string;
+  zonename: string;
+  agreementAccepted?: boolean;
+
+  zones?: Array<Partial<BaseTemplateModel>>;
+}
 
+export const downloadUrl = (template: BaseTemplateModel): string => {
+  const tag = template.tags.find(_ => _.key === TemplateTagKeys.downloadUrl);
 
-@FieldMapper({
-  displaytext: 'displayText',
-  domainid: 'domainId',
-  isdynamicallyscalable: 'isDynamicallyScalable',
-  isextractable: 'isExtractable',
-  isfeatured: 'isFeatured',
-  ispublic: 'isPublic',
-  isready: 'isReady',
-  ostypeid: 'osTypeId',
-  ostypename: 'osTypeName',
-  zoneid: 'zoneId',
-  zonename: 'zoneName',
-})
-export abstract class BaseTemplateModel extends BaseModel implements Taggable {
-  public abstract resourceType: string;
-
-  public path: string;
-
-  public id: string;
-  public account: string;
-  public created: Date;
-  public crossZones: boolean;
-  public displayText: string;
-  public domain: string;
-  public domainId: string;
-  public isDynamicallyScalable: boolean;
-  public isExtractable: boolean;
-  public isFeatured: boolean;
-  public isPublic: boolean;
-  public isReady: boolean;
-  public name: string;
-  public osTypeId: string;
-  public osTypeName: string;
-  public osType: OsType;
-  public size: number;
-  public status: string;
-  public tags: Array<Tag>;
-  public zoneId: string;
-  public zoneName: string;
-  public agreementAccepted?: boolean;
-
-  public zones?: Array<Partial<BaseTemplateModel>>;
-
-  constructor(json) {
-    super(json);
-    this.created = moment(json.created).toDate();
-    this.tags = this.tags || [];
-    this.size = this.size || 0;
+  if (tag) {
+    return tag.value;
   }
+};
 
-  public abstract get isTemplate(): boolean;
+export const sizeInGB = (template: BaseTemplateModel): number => {
+  return Utils.convertToGb(template.size);
+};
 
-  public get sizeInGB(): number {
-    return Utils.convertToGb(this.size);
-  }
+export const isTemplate = (template: BaseTemplateModel): boolean =>
+  template && template['bootable'] !== undefined ? false : true;
 
-  public get downloadUrl(): string {
-    const tag = this.tags.find(_ => _.key === TemplateTagKeys.downloadUrl);
+export const resourceType = (template: BaseTemplateModel) =>
```

Comment:
please consider adding a TemplateType string enum and add a return type to this function